libs/datalayout/build.gradle

```diff
@@ -1,7 +1,6 @@
 plugins {
     id 'openhouse.java-conventions'
     id 'openhouse.hadoop-conventions'
-    id 'openhouse.iceberg-conventions-1.2'
```

Comment:
Why removing this?

Reply:
this convention defines strictly 1.2. that strictly causes issue when used as a dependency in :apps:spark-3.5 which needs iceberg 1.5.

we can also do:
1. replace this line with 1.5 conventions, since this can be overriden by :apps:spark-3.1 since it isn't strictly. this is the same approach you used with services modules to 1.5 and overriding with strictly 1.2
2. exclude the transitive dependency from :apps:spark-3.5, but I would prefer not to do that and use the version resolver instead

Reply:
compiling this module with spark 3.5/iceberg1.5 breaks :apps:spark3.1 

but compiling this module with spark3.1/iceberg1.2 works in :apps:spark3.5 (although unsafe, but this is ok since it won't be productionized until next year)

so leaving this :libs:datalayout as a nonstrict 1.2 is preferred , which requires removing iceberg-conventions-1.2

Reply:
ok with this approach, i'm curious what it would take to build 2 artifacts from this module

Reply:
As an alternative, we can keep 2 versions of this module. But since it is only used internally, we can leverage the auto version upgrade of gradle and exclude the iceberg-spark-runtime in `:apps:spark3.5`. I'm fine with this approach.

Reply:
@teamurko ^ helps to answer your q. we would have two gradle build scripts. one common source code dir (or two code dirs if the source code must diverge significantly between spark3.1/spark3.5 compiled versions)